DBoW2/DBoW2_sample_code.cpp

```diff
@@ -0,0 +1,218 @@
+#include "DBoW2.h"
+#include <opencv2/core.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/features2d.hpp>
+#include <easy/profiler.h>
+
+#include <iostream>
+#include <vector>
+#include <filesystem>
+#include <string>
+#include <memory>
+
+using namespace DBoW2;
+namespace fs = std::filesystem;
+namespace UnitTest
+{
+  using FeatureVector = std::vector<std::vector<cv::Mat>>;
+  using Features = std::vector<cv::Mat>;
+  using FileLUT = std::vector<std::string>;
+}
+
+bool loadDBFeatures(UnitTest::FeatureVector &vfeatures, UnitTest::FileLUT &dbTable, const cv::Ptr<cv::ORB> &orb);
+bool loadVocabulary(OrbVocabulary &voc);
+void changeStructure(const cv::Mat &plain, UnitTest::Features &out);
+bool createDatabase(OrbDatabase &db, const UnitTest::FeatureVector &vfeatures);
+bool loadQueryFeatures(UnitTest::FeatureVector &vfeatures, const cv::Ptr<cv::ORB> &orb);
+bool queryDatabase(OrbDatabase &db, const UnitTest::FeatureVector &vfeatures, UnitTest::FileLUT &queryResultTable, const UnitTest::FileLUT &dbTable);
+bool saveResult(const UnitTest::FileLUT &queryResultTable);
+
+
+// Number of Images to Build Database
+constexpr int NDBIMAGES = 4;
+// Number of Images to Query with Database
+constexpr int NQUERYIMAGES = 1;
+
+
+// Paths 
+fs::path vocPath("../ORBvoc/ORBvoc.txt");
+fs::path dbPath("../demo/Database/");
+fs::path queryPath("../demo/Query/");
+fs::path savePath("../demo/Result/");
+
+
+int main()
+{
+  EASY_MAIN_THREAD;
+  EASY_PROFILER_ENABLE;
+
+  // DataStructures
+  UnitTest::FileLUT queryResultTable;
+  UnitTest::FileLUT dbTable;
+  UnitTest::FeatureVector dbFeatures;
+  UnitTest::FeatureVector queryFeatures;
+  cv::Ptr<cv::ORB> ORB = cv::ORB::create();
+  std::unique_ptr<OrbVocabulary> ptrVocabulary(new OrbVocabulary());

```

Comment:
```suggestion
  auto ptrVocabulary = std::make_unique<OrbVocabulary>();

```

smart pointer를 만들때는 `new`는 사용하지 않습니다.
make_unique, make_shared를 사용합니다

Reply:
다른 코드에도 적용 부탁드립니다!